app/services/invoices/retry_service.rb

```diff
@@ -12,10 +12,12 @@ def call
       return result.not_found_failure!(resource: 'invoice') unless invoice
       return result.not_allowed_failure!(code: 'invalid_status') unless invoice.failed?
 
+      invoice.error_details.tax_error.kept.update_all(deleted_at: Time.current) # rubocop:disable Rails/SkipsModelValidations
       taxes_result = Integrations::Aggregator::Taxes::Invoices::CreateService.call(invoice:, fees: invoice.fees)
 
       unless taxes_result.success?
-        return result.validation_failure!(errors: {tax_error: [taxes_result.error.code]})
+        create_error_detail(taxes_result.error.code)
+        return result.service_failure!(code: 'tax_error', message: taxes_result.error.code)
```

Comment:
We should return `return result.validation_failure!(errors: {tax_error: [taxes_result.error.code]})` here. Even the source of this error is external service, this `RetryService` returns errors directly to the caller. On GQL side and api side we expose only several types of errors. In this case it makes sense to return 422. Please take a look at `execution_error_responder.rb`. Without returning validation error this code would return 500 to the GQL

Reply:
ahaaaa, thank you! 👍 
checking it now